docs/Components.md

```diff
@@ -0,0 +1,235 @@
+
+```tsx
+<GradientBackground colors={["#422443", "#281b34"]} />
+```
+
+### Header
+
+The `Header` component is a component that will appear at the top of your screen. It is used to hold navigation buttons and the screen title.
+
+```tsx
+<Header
+  headerTx="header.title"
+  headerText="Header Title"
+  leftIcon="back"
+  rightIcon="bullet"
+  onLeftPress={() => navigation.goBack()}
+  onRightPress={() => Alert.alert("pressed")}
+  style={{ backgroundColor: "purple" }}
+  titleStyle={{ color: "white" }}
+/>
+```
+
+[Full Header Component Documentation](./Components-Header.md)
+
+### Icon
+
+This is a component that renders an icon.
+
+```tsx
+<Icon
+  icon='back'
+  containerStyle={{backgroundColor: '#fff'}}
+  style={{resizeMode: 'contain'}}
+/>
+```
+
+[Full Icon Component Documentation](./Components-Icon.md)
+
+### Screen
+
+This is a component that renders a screen. It is used to wrap your entire screen, and handles scrolling, [safe areas](https://reactnavigation.org/docs/handling-safe-area/), and keyboard avoiding behavior.
+
+```tsx
+<Screen preset="scroll" >
+  <Header headerTitle="screen"/>
+  // ... content here ...
+</Screen>
+```
+
+[Full Screen Component Documentation](./Components-Screen.md)
+
+### Switch
+
+This is a component that renders a few animated views that can be toggled on and off like a switch. This component handles all the logic of the animation, and all you need to do is style it as you see fit.
+
+```tsx
+<Switch
+  value={switch}
+  onToggle={() => setSwitch(!switch)}
+  style={{ backgroundColor: "purple" }}
+  trackOnStyle={{ backgroundColor: "red" }}
+  trackOffStyle={{ backgroundColor: "blue" }}
+  thumbOnStyle={{ backgroundColor: "red" }}
+  thumbOffStyle={{ backgroundColor: "blue" }}
+/>
+```
+
+[Full Switch Component Documentation](./Components-Switch.md)
+
+### Text
+
+This is an enhanced version of the built-in React Native Text component. It adds internationalization and property presets.
+
+```tsx
+<Text
+  preset="header"
+  tx="welcome.header"
+  txOptions={{
+    name: rootStore.currentUser.name,
+  }}
+  style={$header}
+/>
+```
+
+[Full Text Component Documentation](./Components-Text.md)
+
+### TextField
+
+
+This component renders a View with a [`TextInput`](https://reactnative.dev/docs/textinput) and a text label.
+
+```tsx
+[input, setInput] = useState("");
+const inputRef = useRef();
+<TextField
+  value={input}
+  onChangeText={setInput}
+  labelTx="signup.name"
+  placeholderTx="signup.nameplaceholder"
+  style={$header}
+  inputStyle={$inputStyle}
+  preset="default"
+  forwardedRef={inputRef}
+/>
+```
+[Full Text Component Documentation](./Components-TextField.md)
+
+### Wallpaper
+
+This component renders an image background to a component or a screen.
+
+```tsx
+ <Wallpaper
+  style={{
+    marginHorizontal: 10,
+    paddingVertical: 10,
+  }}
+  backgroundImage={newImage}
+  preset="stretch"
+ />
+```
+[Full Text Component Documentation](./Components-Wallpaper.md)
```

Comment:
Let's add a section at the bottom that talks about custom components.

Mention the component generator:

```
npx ignite-cli generate component MyCustomButton
```